torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -486,6 +486,40 @@ def _mark_manual_sharding(
   return wrap_as_sharded_tensor(t)
 
 
+def enable_manual_sharding(t: Union[torch.Tensor, XLAShardedTensor],
+                           partition_spec: Tuple[Union[Tuple, int, str, None]],
+                           *,
+                           mesh: Mesh = None) -> XLAShardedTensor:
+  """
+  This API enables manual sharding for the given tensor. Manual sharding disables auto sharding proporgation and auto
+  partition for the given tensor and all subsequential tensors that produced by an op that uses the given tensor as
+  input, and therefore allows the user to manually call collectives for the tensor and subsequential tensors. It
```

Comment:
Also just curious - how will we enable collectives in a manual region?

Reply:
XLA cc ops by default should work. Just use it as normal. However, we need to teach our cc ops wrapper to be aware of SPMD mode. So, it will be phase 2 of the mnual sharding.